src/Shared/CertificateGeneration/MacOSCertificateManager.cs

```diff
@@ -388,7 +388,10 @@ protected override void PopulateCertificatesFromStore(X509Store store, List<X509
             // Certs created (or "upgraded") by .NET 7+.
             // .NET 7+ installs the certificate on disk as well as on the user keychain (for backwards
             // compatibility with pre-.NET 7).
-            var onDiskAndKeychain = certsFromDisk.Intersect(certsFromStore, ThumbprintComparer.Instance);
+            // Note that the actual certs we populate need to be the ones from the store location, and
+            // not the version from disk, since we may do other operations with these certs later (such
+            // as exporting) which would fail with crypto errors otherwise.
+            var onDiskAndKeychain = certsFromStore.Intersect(certsFromDisk, ThumbprintComparer.Instance);
```

Comment:
We should probably consider at least starting a list of some of these things for maybe a single CTI scenario for to do some ad hoc verification every once in a while, given that we don't have any test coverage for these kinds of things, (for our future selves)

Reply:
This was actually caught by the WebTools CTI team (see linked issue) as part of their RC 1 pass.

Reply:
Awesome, so we do have coverage, that makes me feel better that they are exercising things then